docs/tests.md

```diff
@@ -0,0 +1,439 @@
+Test waits for the curl output and checks if it contains expected response (401 for required case, 200 for optional).
+
+### Managed Cluster (common machinesets tests)
+Every single test in this section creates a manineset with 2 machines, because of that, this tests
+takes a significant time (>5 m).
+
+All of the test listed below shares common `BeforeEach` and `AfterEach` blocks.
+In `BeforeEach` machineset creation and wait happen (this machineset has 2 desired replicas configured, so 2 machines will be created for each test here).
+In `AfterEach` created machineset is being deleted.
+
+#### Managed cluster should drain node before removing machine resource
+* Parallel: Yes
+* ExecTime: >12m on AWS
+* Recommendations:
+  * Update resource creation helper functions, and add parameters to specify workload names, replicas count, selectors, etc. Now it's almost impossible to say what is going on without looking into these functions.
+  * Specify selectors more strictly for the created workloads, now it's theoretically possible that created workload will be scheduled somewhere else and we could get false positive results.
+  * Add more meaningful comments inside the test code, now it's not easy to figure out what is going on there.
+  * Fix logging here, it is quite noisy, and do flood reports.
+
+This test checks nodes are properly draining during machine deletion and this process also respects PDBs (pod disruption budgets).
+During this test - a replication controller creating with 20 desired replicas, then PDB creating for this workload.
+After the replication controller is ready, machineset marking for deletion. Along with the deletion process test checking that pods amount on the node 
+is going down to 1 and RC has at most one non-ready replica.
+
+Test considered failed in case if more than 1 replica within RC detected as not ready during machineset deletion. That would mean that PDB constraints were violated.
+
+#### Managed cluster should recover from deleted worker machines
+* Parallel: Yes
+* ExecTime: >11m on AWS
+* Recommendations:
+  * Reduce created machines number to 1 for this test, currently there are 2
+  * This test might be possibly removed since it exercises a machineset ability to recreate deleted machines and implemented with envtest
+
+This test takes machines from the machineset which was created in `BeforeEach` block and marks them for deletion,
+then waits for machines to be deleted and recreated by the machineset. Takes quite a while, because 2 machines should be deleted and recreated then.
+
+#### Managed cluster should grow and decrease when scaling different machineSets simultaneously
+* Parallel: Yes
+* ExecTime: >10m on AWS
+* Recommendations:
+  * Reduce created machines number to 1 for this test, currently there are 2
+  * Remove this test, similiar one included into origin suite. Aside of that we running this and other tests which creates and scales MSes in parallel, so we are excercising this functionality
```

Comment:
I think this makes sense to move to a periodic, I think that's been discussed in Mikhails periodic recommendations